sklearn/metrics/_pairwise_distances_reduction.pyx

```diff
@@ -883,7 +884,7 @@ cdef class PairwiseDistancesArgKmin(PairwiseDistancesReduction):
             # We need to recompute distances because we relied on
             # surrogate distances for the reduction.
             self.compute_exact_distances()
-            return np.asarray(self.argkmin_indices), np.asarray(self.argkmin_distances)
+            return np.asarray(self.argkmin_distances), np.asarray(self.argkmin_indices)
```

Comment:
Where does this switch in the return order come from ?
I find it weird that the quantity that is always returned (indices here) is not always at the same place. Depending on some params, the first array returned is not the same. I think it's likely to be the source of errors and bugs at some point

Reply:
`kneighbors` returns values in this order.

Doing otherwise comes with complex boilerplate.

Previous reviews' discussions converged to sticking to the way `kneighbors` return values even if it's not the most intuitive way to return values.

Reply:
This question is asked often enough. Can we leave a comment briefly explaining that we have this order to be consistent with `kneighbors`?

(I also thing the order is weird)

Reply:
Addressed in b6e6f3d38fb2f931cece6d87c4c569f977941dbf.